[🎉deploy] CI/CD 테스트 진행

junggyo1020 opened the pull request:
# ⚙️ ISSUE
- closed #

# 📄 Work Description
- CI/CD 배포 자동화 테스트를 진행합니다.

## 진행상황
###  CI & CD 로직 정상적으로 실행 확인
<img width="602" alt="image" src="https://github.com/user-attachments/assets/d5a10c55-f267-4fca-9565-17170420d427" />

### 컨테이너 정상적으로 실행 확인
<img width="1047" alt="image" src="https://github.com/user-attachments/assets/9139af74-4953-4021-9a90-5701066c1be6" />


# ✅ PR check list
- [x] Reviewers
- [x] Assignees
- [x] Labels

jsoonworld committed:
feat(GlobalErrorCode) : 500번대 글로벌 에러 코드 정의

서버 내부 오류 및 게이트웨이 관련 예외를 처리하기 위한 전역 에러 코드 열거형 구현.
공통 접두사를 포함한 에러 메시지 형식을 제공함.

jsoonworld committed:
feat(GlobalExceptionHandler) : 500번대 예외에 대한 전역 예외 처리 로직 구현

BaseException 및 주요 시스템 예외(501, 502, 503, 504 등)를 처리하는 전역 예외 핸들러 정의.
각 예외에 대해 GlobalErrorCode를 매핑하여 일관된 에러 응답을 반환하도록 구성함.

jsoonworld committed:
feat(SuccessCode) : 성공 응답 코드 인터페이스 정의

성공 응답에 사용될 HTTP 상태 코드와 메시지를 제공하는 공통 인터페이스 생성.

jsoonworld committed:
feat(SuccessResponse) : 성공 응답 DTO 구현 및 상태 코드 정수 변환 처리

성공 결과를 클라이언트에 전달하기 위한 공통 응답 객체 정의.
HttpStatus를 int로 변환하여 응답 형식을 표준화하고, 결과 데이터는 null일 경우 제외되도록 설정함.

jsoonworld committed:
del(temp) : 사용하지 않는 파일 삭제

jsoonworld committed:
feat(UserErrorCode) : 사용자 이름 제약 및 에러 코드 정의

- 이름 null 허용 불가, 최대 길이 초과, 형식 오류에 대한 에러 코드 추가
- 공통 메시지 접두사 PREFIX 적용
- 이름 최대 길이 상수 NAME_MAX_LENGTH 정의

jsoonworld committed:
feat(UserException) : 사용자 도메인 전용 예외 클래스 구현

- UserErrorCode를 인자로 받아 BaseException을 상속하는 예외 정의
- 사용자 관련 예외 처리를 도메인 단위로 구분 가능하도록 구성

jsoonworld committed:
refactor(UserName) : 사용자 이름 검증 시 UserException 사용하도록 변경

- 기존 IllegalArgumentException을 UserException으로 대체하여 도메인 예외 통일
- UserErrorCode를 활용해 에러 메시지 관리 일원화
- 이름 검증 로직에서 null, 길이, 형식 예외를 도메인 기반으로 처리

jsoonworld committed:
test(UserNameTest) : UserException 기반 예외 메시지 검증 테스트로 수정

- 기존 IllegalArgumentException 검증을 UserException으로 변경
- UserErrorCode의 메시지를 기반으로 예외 메시지 검증 수행
- 사용자 이름 검증 관련 테스트 일관성 유지

jsoonworld committed:
feat(UserErrorCode) : 푸시 토큰 null 검사 에러 코드 구현

푸시 토큰이 null일 수 없다는 검증을 위한 에러 코드를 UserErrorCode에 추가했습니다.

jsoonworld committed:
feat(User) : PushToken VO 적용을 위한 필드 타입 변경

기존 String 타입의 token 필드를 PushToken VO로 변경하고, @Embedded 및 @AttributeOverride를 적용하여 도메인 주도 설계 기반의 값 객체 구조를 반영했습니다.

jsoonworld committed:
feat(PushToken) : PushToken VO 구현

푸시 토큰 값을 표현하는 PushToken VO를 구현했습니다.
null 값 검증 로직을 포함하고 있으며, 추후 토큰 유효성에 대한 비즈니스 로직과 테스트를 추가할 예정입니다.

jsoonworld committed:
test(PushToken) : PushToken VO 테스트 코드 작성

PushToken 값 객체에 대한 단위 테스트를 추가했습니다.
- 유효한 토큰으로 객체 생성 성공
- null 토큰에 대한 예외 처리 검증

jsoonworld committed:
refactor(UserNameTest) : UserName 테스트 패키지 위치 이동

jsoonworld committed:
feat(UserErrorCode) : 푸시 알림 관련 에러 코드 구현

- 푸시 토큰 null 허용 불가 에러 코드 추가
- 푸시 상태가 ON, OFF 외의 값일 경우의 에러 코드 추가
- 푸시 알림이 이미 활성화/비활성화된 상태에 대한 예외 코드 추가

jsoonworld committed:
feat(User) : 푸시 알림 상태 관리 기능 및 필드 추가

- PushNotificationStatus VO 필드 추가 및 ENUM 매핑 적용
- 푸시 알림 수신 여부 확인 메서드 canReceivePushNotification() 구현
- 푸시 알림 활성화 및 비활성화 메서드 추가

jsoonworld committed:
feat(PushNotificationStatus) : 푸시 알림 상태 값과 전환 로직 구현

- ENABLED, DISABLED 상태를 나타내는 Enum 정의
- 문자열 기반 상태 파싱을 위한 of(), isValid() 메서드 구현
- 상태 전환 시 중복 전환 방지를 위한 예외 처리 로직 추가
- 알림 수신 가능 여부 확인 메서드 canReceiveNotification() 구현

jsoonworld committed:
test(PushNotificationStatusTest) : 푸시 알림 상태 값 검증 테스트 구현

- of(), isValid() 메서드의 유효/무효 입력에 대한 테스트 추가
- 상태 전환 시 중복 호출에 대한 예외 처리 테스트 추가
- 상태별 알림 수신 가능 여부 테스트 및 상태 전환 테스트 포함
- 성공/실패 케이스를 명확히 나누어 가독성 향상

JungYoonShin commented on .github/workflows/DEV-CI.yml:
```diff
@@ -0,0 +1,42 @@
+          distribution: 'temurin'
+          java-version: '21'
+
+      # # Gradle caching - 빌드 시간 향상
+      # - name: Gradle Caching
+      #   uses: actions/cache@v3
+      #   with:
+      #     path: |
+      #       ~/.gradle/caches
+      #       ~/.gradle/wrapper
+      #     key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+      #     restore-keys: |
+      #       ${{ runner.os }}-gradle-
+
+      # Gradle build - 테스트 없이 gradle 빌드
```
이거 주석처리한 이유가 먼가용?? (단순 궁금ㅎㅎ)

junggyo1020 replied:
해당 빌드 캐싱이 자주 빌드해야하는 상황에서 오히려 시간이 걸리더라구요! 이유를 파악하고 이후 반영을 고려하고 있습니다:)

JungYoonShin commented on HELP.md:
이 파일은 모져?!?! 잘못 올라왓나

junggyo1020 replied:
start.spring.io에서 최근에 gradle파일을 추가하는 경우 자동으로 생성되는 것 같아요! 이부분은 반영되지 않도록 gitignore에 추가해야겠네요!! 의견 감사합니다 ㅎㅎ

JungYoonShin reviewed (approved):
수고하셨슴동!

jsoonworld commented:
고생 많으셨습니다~! 😊

추가로 `deploy` 파일에 도커 이미지 삭제 로직도 포함되었는지 한 번 확인 부탁드릴게요.
예전에 기존 서버에서는 이 부분이 빠져 있어서 간헐적으로 문제가 생겼던 적이 있었거든요.

그리고 서버가 중간에 죽고 다시 올라올 때처럼,
예외 상황이 생길 수 있는 흐름도 한 번 더 점검해보시면 더 안정적으로 동작할 것 같아요 🙏

jsoonworld committed:
feat(AccountStatus) : 사용자 계정 상태 enum 및 유틸 메서드 구현

- LOGGED_IN, LOGGED_OUT, WITHDRAWN 상태를 나타내는 enum 정의
- 문자열 파싱 메서드 of(), isValid() 구현
- 로그인, 로그아웃, 탈퇴 상태 확인을 위한 메서드 추가
- 유효하지 않은 상태에 대해 UserException 발생 처리

jsoonworld committed:
feat(UserErrorCode) : 계정 상태 값 유효성 검사용 에러 코드 추가

- 유효하지 않은 계정 상태 입력 시 발생하는 INVALID_ACCOUNT_STATUS 예외 코드 정의
- 허용된 상태 값: active, inactive, withdrawn

jsoonworld committed:
feat(User) : 계정 상태 필드 AccountStatus 적용 및 관련 메서드 추가

- 기존 State 필드 제거 및 AccountStatus enum 기반으로 변경
- 계정 활성 상태 판단 메서드 isActiveUser() 구현
- 탈퇴 여부를 AccountStatus가 직접 판단하도록 역할 위임

jsoonworld committed:
refactor(AccountStatus) : 외부 표현 값 추가 및 구조 개선

- 각 상태에 외부 노출용 문자열 값(value) 필드 추가 (active, inactive, withdrawn)
- Jackson 직렬화/역직렬화를 위한 @JsonCreator 및 @JsonValue 어노테이션 적용
- 문자열 입력 파싱을 value 기준으로 처리하도록 from(), isValid() 메서드 개선
- 의미 기반 메서드 isActive(), isInactive(), isWithdrawn() 등으로 명확성 향상

jsoonworld committed:
test(AccountStatusTest) : 계정 상태 enum 기능 테스트 추가

- from() 메서드의 유효/무효 입력에 대한 예외 처리 테스트 구현
- isValid() 메서드의 검증 테스트 추가
- isActive(), isInactive(), isWithdrawn() 메서드 테스트로 상태 판별 기능 확인
- value() 메서드를 통한 직렬화 값 반환 검증

jsoonworld committed:
refactor(PushNotificationStatus) : 외부 표현 값 추가 및 JSON 직렬화 대응

- 각 상태에 외부 노출용 문자열 값(value) 필드 추가 ("enabled", "disabled")
- @JsonCreator, @JsonValue 적용으로 직렬화/역직렬화 대응
- from(), isValid() 메서드가 외부 입력 값 기준으로 동작하도록 수정
- 내부 Enum 이름과 외부 표현을 분리하여 API 유연성과 유지보수성 향상

jsoonworld committed:
test(PushNotificationStatusTest) : 외부 문자열 기반 from() 메서드 및 직렬화 값 테스트로 수정

- 기존 of() 메서드 기반 테스트를 from() 기반으로 변경
- Json 직렬화를 고려한 외부 표현값("enabled", "disabled") 테스트로 개선
- isValid() 메서드에 대해서도 대소문자 및 value 기준 입력 검증 테스트 적용
- 테스트 메서드명 및 DisplayName도 동작 방식에 맞게 리네이밍

jsoonworld committed:
feat(UserErrorCode) : 인증 타입 검증 실패 시 예외 코드 추가

- 허용되지 않은 인증 타입 입력에 대한 INVALID_AUTH_TYPE 예외 코드 정의
- 유효한 인증 타입: apple, kakao

jsoonworld committed:
refactor(AuthType) : 외부 표현 값 및 예외 처리 로직 추가

- 인증 타입 enum에 외부 표현용 문자열 값(value) 필드 추가 ("apple", "kakao")
- from(), isValid() 메서드를 통해 문자열 파싱 및 유효성 검사 지원
- 유효하지 않은 입력에 대해 UserException 발생
- @JsonCreator, @JsonValue 적용으로 직렬화/역직렬화 대응
- 기존 단순 enum(AuthType) 제거 및 도메인 VO로 대체

jsoonworld committed:
chore(User) : 도메인 VO 패키지 import 구문 정리

- 동일 패키지 내 VO 클래스들을 wildcard(*) import로 정리하여 가독성 개선

jsoonworld committed:
test(AuthTypeTest) : 인증 타입 enum 기능 테스트 추가

- from() 메서드의 유효/무효 입력에 대한 예외 처리 테스트 구현
- isValid() 메서드에 대한 유효성 검증 테스트 추가
- value() 메서드를 통한 직렬화용 문자열 반환 값 확인
- 대소문자 구분 없이 작동하는 enum 매핑 로직 검증

jsoonworld committed:
refactor(User) : @AttributeOverride name 필드를 내부 필드명(value) 기준으로 수정

- UserName, PushToken 임베디드 타입에서 필드명이 'value'로 변경된 구조 반영
- JPA 매핑 오류 방지를 위해 @AttributeOverride(name = "value", ...)로 수정

jsoonworld committed:
refactor(PushToken) : 내부 필드명 value로 통일 및 매핑 일관성 확보

- 기존 필드명 token → value로 변경하여 임베디드 타입 구조와 통일
- 생성자, 검증 메서드, 반환 메서드 모두 value 기준으로 수정
- JPA @AttributeOverride와의 일관성 및 도메인 VO 설계 통일성 확보

jsoonworld committed:
refactor(UserName) : 내부 필드명을 value로 변경하여 VO 구조 통일

- 기존 필드명 name → value로 변경하여 PushToken 등 다른 VO와 일관성 확보
- 생성자, 검증 메서드, 출력 메서드 모두 value 기반으로 수정
- @AttributeOverride(name = "value", ...) 설정과의 매핑 일치
- 도메인 VO 간 구조 통일로 유지보수성과 확장성 향상

junggyo1020 commented:
@jsoonworld 
우선 docker image를 삭제하는 로직은 반영을 했습니다!
발생할 수 있는 예외상황에 대해서는 조금 더 생각을 해보아야겠네요! 좋은 의견 감사합니다:)

jsoonworld committed:
merge(global) : 공통 예외 처리 기능 구현

[✨ feat] 공통 예외 처리 기능 구현

jsoonworld committed:
merge(UserName) : UserName에 커스텀 예외처리 적용 

[✨ feat] UserName에 커스텀 예외처리 적용

jsoonworld committed:
merge(PushToken) : FCM 토큰 VO 구현 

[✨ feat] FCM 토큰 VO 구현

jsoonworld committed:
merge(PushNotificationStatus) : 푸시 알람 설정 VO 구현 

[✨ feat] 푸시 알람 설정 VO 구현

jsoonworld committed:
merge(AccountStatus) : 계정 상태 VO 구현

[✨ feat] 계정 상태 VO 구현

junggyo1020 committed:
Update issue templates

jsoonworld committed:
merge(PushNotificationStatus) : 푸시 알림 상태 enum에 외부 표현 값 추가 및 구조 개선

[♻️ refactor] 푸시 알림 상태 enum에 외부 표현 값 추가 및 구조 개선

jsoonworld committed:
merge(AuthType) : 소셜 로그인 타입 VO 구현 

[✨ feat] 소셜 로그인 타입 VO 구현

jsoonworld committed:
merge(value):  VO 필드명을 value로 일관성 있게 정리 

[♻️ refactor] VO 필드명을 value로 일관성 있게 정리

thingineeer committed:
Merge branch 'develop' into feat/#25

thingineeer committed:
Merge branch 'develop' into refactor/#22

thingineeer committed:
Merge branch 'develop' into feat/#13

thingineeer committed:
Merge branch 'develop' into feat/#15

thingineeer committed:
Merge branch 'develop' into feat/#17

thingineeer committed:
Merge branch 'develop' into feat/#19

thingineeer committed:
Merge branch 'develop' into feat/#21

thingineeer committed:
Merge branch 'refactor/#22' into feat/#25

thingineeer committed:
Merge branch 'feat/#21' into feat/#25

thingineeer committed:
Merge branch 'feat/#19' into feat/#25

thingineeer committed:
Merge branch 'feat/#17' into feat/#25

thingineeer committed:
Merge branch 'feat/#15' into feat/#25

thingineeer committed:
Merge branch 'feat/#13' into feat/#25

jsoonworld committed:
merge(develop) : feature - > develop

Feat/#25

jsoonworld committed:
refactor(GlobalExceptionHandler) : 불필요한 예외 핸들러 제거

BaseException과 Exception 처리만으로 충분하므로,
UnsupportedOperationException, HttpMediaTypeNotSupportedException,
SocketTimeoutException에 대한 별도 핸들러를 제거함.

jsoonworld committed:
refactor(UserName) : 이름 유효성 검증 순서 조정

jsoonworld committed:
refactor(UserErrorCode) : 사용자 이름 허용 문자 설명 문구 개선

- INVALID_USER_NAME 메시지에 공백 허용 여부를 명확히 표현

jsoonworld committed:
refactor(FcmToken) : PushToken에서 FcmToken으로 클래스명 변경 및 예외 코드 정리

- 클래스명 PushToken → FcmToken 으로 명확히 표현
- 생성자, 팩토리 메서드 등 관련 메서드명 일괄 수정
- 예외 코드도 FCM_TOKEN_NOT_NULL 로 변경

jsoonworld committed:
refactor(User) : PushToken 필드명을 FcmToken으로 변경

- @Embedded 필드 타입을 PushToken → FcmToken으로 수정
- 클래스명 변경에 따른 필드 타입 일관성 유지

jsoonworld committed:
test(FcmTokenTest) : PushTokenTest 클래스명을 FcmTokenTest로 변경

- FcmToken 클래스명 변경에 따라 테스트 클래스 및 내부 테스트 대상 수정
- 예외 메시지 검증 시 FCM_TOKEN_NOT_NULL 에 맞춰 assert 문 수정

jsoonworld committed:
refactor(UserErrorCode) : FCM 토큰용 예외 코드로 명확히 분리

- PUSH_TOKEN_NOT_NULL → FCM_TOKEN_NOT_NULL 로 수정하여 도메인 명확성 확보

jsoonworld committed:
test(FcmTokenTest) : 테스트 클래스 설명 문구 FCM 기준으로 수정

- @DisplayName 어노테이션을 "푸시 토큰 테스트" → "FCM 토큰 테스트"로 변경

jsoonworld committed:
feat(ScrapErrorCode) : 스크랩 도메인 예외 코드 구현

스크랩 상태에 대한 유효성 검사 및 중복 상태 예외를 처리하기 위한 ScrapErrorCode 열거형을 구현하였습니다.
- scrapped/unscrapped 외 상태에 대한 예외 정의
- 이미 스크랩된 상태 또는 스크랩되지 않은 상태에 대한 예외 정의
- [SCRAP ERROR] prefix 포함한 에러 메시지 포맷 구성

jsoonworld committed:
feat(ScrapException) : 스크랩 예외 처리용 커스텀 예외 클래스 구현

ScrapErrorCode를 인자로 받아 스크랩 관련 예외를 처리할 수 있도록 ScrapException을 정의하였습니다.
- 공통 예외 상위 클래스(BaseException) 상속
- 스크랩 도메인 전용 예외 처리 지원

jsoonworld committed:
feat(Scrap) : Scrap 엔티티 구현 및 기존 Scraps 클래스 제거

- 사용자(User)와 연관된 스크랩 정보를 저장하는 Scrap 엔티티 정의
- 스크랩 상태 관리(SCRAPPED/UNSCRAPPED)를 위한 ScrapStatus Enum과 연동
- 스크랩 생성 정적 팩토리 메서드 `of(User user)` 제공
- 상태 전이 메서드(scrap, unscrap) 및 상태 확인 메서드(isScrapped, isUnscrapped) 추가
- 불필요한 Scraps 클래스 제거

jsoonworld committed:
feat(ScrapStatus) : 스크랩 상태 Enum 및 상태 전이 로직 구현

- SCRAPPED, UNSCRAPPED 두 가지 상태를 정의하는 ScrapStatus Enum 구현
- 문자열 입력으로부터 Enum 매핑 기능(from) 및 유효성 검증 메서드(isValid) 제공
- 동일 상태로의 중복 전이를 방지하는 scrap(), unscrap() 메서드 구현
- 상태 확인 메서드(isScrapped, isUnscrapped) 및 Jackson 직렬화/역직렬화 설정 추가
- 잘못된 상태 입력 또는 중복 전이 시 ScrapException 발생 처리

jsoonworld committed:
refactor(User) : Scraps 리스트를 Scrap 리스트로 수정

jsoonworld committed:
test(ScrapStatus) : 스크랩 상태 Enum에 대한 단위 테스트 구현

- from() 메서드의 유효/무효 입력값 처리 검증
- scrap(), unscrap() 전이 메서드 정상 동작 및 예외 케이스 테스트
- isScrapped(), isValid() 등의 유틸성 메서드에 대한 테스트 포함
- 대소문자 구분 없이 상태 변환 가능함을 검증

jsoonworld committed:
feat(ScrapErrorCode) : 스크랩 상태 null 검증용 예외 코드 추가

- ScrapStatus 입력값이 null인 경우를 명확히 처리하기 위해 SCRAP_STATUS_CANNOT_BE_NULL 코드 추가
- from() 메서드 내 null 체크와 연계하여 예외 메시지 명확화

jsoonworld committed:
feat(ScrapStatus) : from() 메서드에 null 입력값 검증 로직 추가

- 입력값이 null일 경우 ScrapStatus.from()에서 NPE 방지를 위해 명시적 검증 로직 추가
- ScrapException을 통해 SCRAP_STATUS_CANNOT_BE_NULL 예외 코드 반환
- null 검증 책임을 분리한 validateNotNull() 메서드로 구현

jsoonworld committed:
test(ScrapStatusTest) : null 입력에 대한 예외 및 유효성 테스트 추가

- ScrapStatus.from(null) 호출 시 ScrapException 발생 여부 테스트
- isValid(null) 호출 시 false 반환 테스트 추가
- null 입력값 처리 로직에 대한 테스트 커버리지 보완

junggyo1020 committed:
[🔬 setting] Firebase Admin SDK 의존성 추가

junggyo1020 committed:
[🔬 setting] yml에 저장한 fcm-service-key를 가져오기 위한 properties 설정 클래스 추가

junggyo1020 committed:
[🔬 setting] FirebaseApp 초기화하는 fcm 구성 클래스 추가

junggyo1020 committed:
[🔬 setting] configuration 파일을 spring에서 읽어오기 위한 Enable 어노테이션 추가

jsoonworld committed:
feat(MessageErrorCode) : 메시지 포맷팅 예외 코드 정의

- 메인 메시지 null 여부 및 유효성 검증 에러 추가
- 메시지 포맷 시 파라미터 누락 및 값 누락에 대한 에러 정의
- 공통 인터페이스 ErrorCode 구현

jsoonworld committed:
feat(MessageException) : 메시지 도메인 전용 예외 클래스 구현

- 메시지 관련 예외 처리를 위한 MessageException 클래스 정의
- BaseException 상속 및 MessageErrorCode 기반 생성자 구현

jsoonworld committed:
feat(AbstractMessageTemplate) : 메시지 포맷팅 공통 로직 구현

- 템플릿 내 플레이스홀더를 파라미터로 치환하는 포맷 메서드 정의
- 포맷팅 필요 여부에 따라 파라미터 검증 및 치환 수행
- 누락된 파라미터나 값이 있을 경우 MessageException 발생
- MessageTemplate 인터페이스 구현을 위한 추상 클래스 설계

jsoonworld committed:
feat(MainMessage) : 메인 메시지 템플릿 enum 정의

- 사용자에게 보여줄 메인 메시지를 Enum 형태로 정의
- 각 항목은 포맷팅이 필요 없는 고정 메시지로 구성
- AbstractMessageTemplate 상속을 통해 포맷팅 기능 확장

jsoonworld committed:
feat(MessageTargetType) : 메시지 대상 사용자 타입 enum 정의

- 메시지 전송 대상 구분을 위한 타입 정의
- 스크랩한 사용자(SCRAPPED_USER)와 전체 사용자(ALL_USERS) 구분

jsoonworld committed:
feat(MessageTemplateType) : 메시지 템플릿 조합 타입 정의

- 메인/서브 메시지 및 타겟 유형을 조합한 템플릿 타입 정의
- 메시지 종류별로 메인 메시지, 서브 메시지, 대상 사용자 연결
- 각 템플릿 타입에서 메시지 포맷 메서드(main, sub) 제공

jsoonworld committed:
feat(SubMessage) : 서브 메시지 템플릿 enum 정의

- 사용자 이름을 포함하는 메시지 본문(SubMessage) 템플릿 정의
- 포맷팅이 필요한 메시지로 구성되어 파라미터 기반 치환 수행
- AbstractMessageTemplate 상속으로 포맷팅 로직 재사용

jsoonworld committed:
refactor(Message) : 메시지 도메인 클래스 이름 변경 및 정리

- Messages → Message로 클래스명 단수형으로 변경하여 의미 명확화
- 동일한 필드(mainMessage, subMessage)는 유지
- 기존 Messages 클래스 제거 및 새로운 Message 클래스 적용

jsoonworld committed:
feat(MessageTemplate) : 메시지 템플릿 포맷 인터페이스 정의

- 메시지 문자열을 반환하는 value 메서드 정의
- 포맷팅 필요 여부를 판단하는 needsFormatting 메서드 정의
- 파라미터 기반 포맷을 수행하는 format 메서드 포함
- AbstractMessageTemplate, MainMessage, SubMessage 등에서 구현 예정

jsoonworld committed:
refactor(Notifications) : 메시지 참조 타입 Messages → Message로 변경

- Messages 클래스명을 Message로 변경함에 따라 필드 타입 및 import 구문 수정
- 관련 JPA 매핑 어노테이션 유지 (cascade, orphanRemoval 등)

jsoonworld committed:
test(MessageTemplateTypeTest) : 메시지 템플릿 포맷 테스트 추가

- MessageTemplateType의 main/sub 메시지 포맷 정상 동작 테스트
- 파라미터 누락, 잘못된 키, null 값 등에 대한 예외 케이스 검증
- 포맷이 불필요한 메인 메시지와 여분 파라미터 무시 동작 확인
- AbstractMessageTemplate의 중복 플레이스홀더 처리 테스트 포함

jsoonworld committed:
feat(Message) : 메시지 도메인에 템플릿 타입 및 포맷 결과 필드 추가

- messageTemplateType 필드 추가 및 Enum 매핑 처리
- formattedMainMessage, formattedSubMessage 필드 추가
- MessageTemplateType 기반 메시지 생성을 위한 of 정적 메서드 구현
- 동일 템플릿 타입 비교를 위한 isSameType 메서드 추가

jsoonworld committed:
test(MessageTest) : 메시지 도메인 생성 및 포맷 테스트 추가

- MessageTemplateType을 기반으로 Message 객체 생성 기능 테스트
- main/sub 메시지 포맷 정상 동작 여부 검증
- 포맷 파라미터 누락, null 값, 잘못된 키 등의 예외 케이스 검증
- isSameType 메서드를 통한 템플릿 타입 비교 기능 테스트

jsoonworld committed:
merge(global) :  글로벌 예외 처리 개선 

[♻️ refactor] 글로벌 예외 처리 개선

jsoonworld committed:
merge(UserName) : UserName 검증 개선

[♻️ refactor] UserName 검증 개선

jsoonworld committed:
merge(PushToken) : PushToken 네이밍 변경

[♻️ refactor] PushToken 네이밍 변경

jsoonworld committed:
merge(ScrapStatus) : 스크랩 상태를 나타내는 ENUM 구현

[✨ feat] 스크랩 상태를 나타내는 ENUM 구현

jsoonworld committed:
Merge branch 'feat/#40' into feat/#45

jsoonworld committed:
Merge branch 'feat/#43' into feat/#45

jsoonworld committed:
Merge branch 'feat/#38' into feat/#45

junggyo1020 committed:
[✒️ comment] 충분히 이해가능하여 불필요한 주석 제거

junggyo1020 committed:
[✅ chore] Encoding 관련 설정 정보를 상수 클래스로 분리

junggyo1020 committed:
[✅ chore] 로그 메세지를 별도의 핸들러 클래스로 관리하도록 수정

junggyo1020 committed:
[✅ chore] 로그 메세지를 별도의 핸들러 클래스로 관리하도록 수정

junggyo1020 committed:
Merge pull request #41 from teamterning/feat/#38

[✨ feat] FCM 환경설정 및 JSON Key 관리 환경 구축

jsoonworld committed:
feat(FcmErrorCode) : FCM 전송 실패 코드 정의

FCM 메시지 전송 실패에 대한 에러 코드를 FcmErrorCode enum으로 정의하였습니다.
- HttpStatus.INTERNAL_SERVER_ERROR 상태와 함께 실패 메시지 제공
- 공통 인터페이스 ErrorCode 구현

jsoonworld committed:
feat(FcmException) : FCM 전송 실패 예외 클래스 구현

FcmErrorCode를 활용한 FCM 전송 실패 예외 FcmException을 정의하였습니다.
- BaseException을 상속받아 공통 예외 처리 방식 유지
- FcmErrorCode 기반으로 에러 메시지 및 상태 전달

jsoonworld committed:
feat(FcmTokenValidator) : FCM 토큰 만료 검증 인터페이스 정의

FCM 토큰이 만료되었는지 여부를 판단하는 FcmTokenValidator 인터페이스를 정의하였습니다.
- FcmToken 도메인 객체 기반의 만료 여부 검증 메서드 선언
- 다양한 구현체 확장을 위한 계약 역할 수행

jsoonworld committed:
feat(FirebaseFcmTokenValidator) : FCM 토큰 만료 검증 로직 구현

FCM 서버에 검증 메시지를 전송하여 FCM 토큰이 만료되었는지를 확인하는 FirebaseFcmTokenValidator 구현체를 추가하였습니다.
- MessagingErrorCode.UNREGISTERED 에러 시 만료로 간주
- 예외 발생 시 FcmException으로 변환하여 공통 에러 처리 유지

jsoonworld committed:
feat(UserController) : FCM 토큰 재발급 필요 여부 확인 API 구현

사용자의 FCM 토큰이 만료되었는지 여부를 확인하는 API를 UserController에 구현하였습니다.
- POST /api/v1/users/fcm-tokens/reissue-required 엔드포인트 추가
- 요청 바디로 FcmTokenReissueRequiredRequest 수신
- 결과는 FcmTokenReissueRequiredResponse로 반환하며, 성공 응답 래핑 처리

jsoonworld committed:
feat(FcmTokenValidateService) : FCM 토큰 재발급 필요 여부 확인 서비스 인터페이스 정의

사용자의 FCM 토큰 재발급 필요 여부를 확인하는 로직을 담당할 FcmTokenValidateService 인터페이스를 정의하였습니다.
- 요청 DTO를 받아 응답 DTO로 결과 반환
- 서비스 로직의 확장성과 테스트 용이성을 고려한 구조

jsoonworld committed:
feat(FcmTokenValidateServiceImpl) : FCM 토큰 재발급 필요 여부 확인 서비스 구현

사용자의 FCM 토큰이 만료되었는지 확인하는 FcmTokenValidateServiceImpl 클래스를 구현하였습니다.
- UserRepository를 통해 사용자 조회
- FcmTokenValidator를 통해 토큰 만료 여부 검증
- 결과를 FcmTokenReissueRequiredResponse로 반환
- 사용자 없을 시 USER_NOT_FOUND 예외 처리

jsoonworld committed:
feat(UserService) : FCM 토큰 재발급 필요 여부 확인 로직 위임

UserService에서 FcmTokenValidateService를 사용해 FCM 토큰 만료 여부 확인 기능을 위임 처리하였습니다.
- FcmTokenReissueRequiredRequest를 받아 서비스 호출
- FcmTokenReissueRequiredResponse 반환

jsoonworld committed:
feat(UserErrorCode) : USER_NOT_FOUND 에러 코드 정의

사용자 조회 실패 시 예외 처리를 위한 USER_NOT_FOUND 에러 코드를 추가하였습니다.
- HttpStatus.NOT_FOUND 상태와 메시지 구성
- 공통 에러 코드 인터페이스 구현을 위한 포맷 유지

jsoonworld committed:
feat(UserSuccessCode) : FCM 토큰 재발급 여부 확인 성공 코드 정의

FCM 토큰 재발급 필요 여부를 성공적으로 응답했을 때 사용할 UserSuccessCode를 추가하였습니다.
- HTTP 상태는 200 OK
- 메시지: "FCM 토큰 재발급 여부를 성공적으로 확인했습니다."

jsoonworld committed:
feat(User) : FCM 토큰 만료 여부 검증 메서드 추가

User 도메인에 FCM 토큰 만료 여부를 FcmTokenValidator를 통해 확인하는 isFcmTokenExpired 메서드를 추가하였습니다.
- token 필드의 위임 메서드 형태로 구현
- User 내부에서 FCM 토큰 유효성 검증 가능하도록 개선

jsoonworld committed:
feat(UserRepository) : 사용자 조회를 위한 JPA 리포지토리 인터페이스 정의

User 엔티티에 대한 기본 CRUD 작업을 수행할 수 있도록 JpaRepository를 상속한 UserRepository 인터페이스를 정의하였습니다.
- Long 타입의 사용자 ID 기반 조회 지원
- @Repository 어노테이션으로 스프링 빈으로 등록

jsoonworld committed:
feat(FcmToken) : FCM 토큰 만료 여부 위임 메서드 추가

FcmTokenValidator를 이용해 토큰 만료 여부를 확인할 수 있는 isExpiredWith 메서드를 FcmToken VO에 추가하였습니다.
- 외부 유효성 검증 전략을 위임받아 처리 가능
- 테스트 및 검증을 위한 TODO 주석 추가
- value 필드 중복 반환 제거로 정리

jsoonworld committed:
feat(FcmTokenReissueRequiredRequest) : FCM 토큰 재발급 필요 여부 요청 DTO 정의

사용자의 FCM 토큰 재발급 필요 여부를 확인하기 위한 요청 DTO를 정의하였습니다.
- userId를 필드로 포함
- of 정적 팩토리 메서드를 통해 객체 생성 가능하도록 구성

jsoonworld committed:
feat(FcmTokenReissueRequiredResponse) : FCM 토큰 재발급 필요 여부 응답 DTO 정의

FCM 토큰 재발급이 필요한지를 클라이언트에 전달하기 위한 응답 DTO를 정의하였습니다.
- reissueRequired 필드를 포함
- 정적 팩토리 메서드 of 제공

jsoonworld committed:
del(TerningApplicationTests) : 단위 테스트 작성을 위한 기본 테스트 클래스 제거

불필요한 contextLoads 메서드만 포함된 기본 생성 테스트 클래스 TerningApplicationTests를 삭제하였습니다.
- 불필요한 통합 테스트 실행을 방지하고 명확한 단위 테스트 환경 구성 목적

jsoonworld committed:
test(FakeFcmTokenValidator) : FCM 토큰 만료 여부 테스트용 구현체 추가

단위 테스트에서 FCM 토큰 만료 여부를 제어하기 위한 FakeFcmTokenValidator 클래스를 추가하였습니다.
- FcmTokenValidator 인터페이스 구현
- setExpired 메서드를 통해 만료 여부를 임의로 설정 가능
- 테스트 독립성과 예측 가능성 확보

jsoonworld committed:
test(FcmTokenValidateServiceImplTest) : FCM 토큰 만료 여부 검증 서비스 단위 테스트 추가

FcmTokenValidateServiceImpl에 대한 단위 테스트를 작성하였습니다.
- FakeFcmTokenValidator를 사용해 만료 여부를 제어
- 토큰이 만료된 경우 true, 그렇지 않으면 false를 반환하는지 확인
- 존재하지 않는 사용자 ID에 대해 예외가 발생하는지 검증
- @DisplayName, @Nested를 활용해 테스트 가독성 향상

jsoonworld committed:
test(UserRepositoryTest) : 사용자 도메인 단위 테스트를 위한 인메모리 리포지토리 구현

단위 테스트에서 UserRepository를 대체할 수 있는 인메모리 기반의 UserRepositoryTest 클래스를 구현하였습니다.
- id 자동 생성 및 필드 직접 주입을 통해 테스트 가능하도록 구성
- save, findById, delete 등 기본 CRUD 기능 구현
- Spring Data JPA의 JpaRepository 인터페이스 구현 메서드 중 필요한 부분만 처리
- 실제 DB에 의존하지 않고 사용자 도메인 테스트 가능

jsoonworld committed:
refactor(FcmTokenValidateServiceImpl): FCM 재발급 판단 로직 개선

jsoonworld committed:
test(FcmTokenValidateServiceImplTest): 테스트 코드 리팩토링 및 변수 정리

- `setUp()`에서 유저 객체를 더 명확히 구분
- `userId`를 잘못된 변수로 재사용하던 부분 수정
- `FcmTokenReissueRequiredRequest` 중복 코드를 하나로 통합
- 잘못된 변수명 수정 및 불필요한 코드 제거
- 테스트의 예외 처리 및 반환 값 검증 부분 수정

jsoonworld committed:
Merge branch 'develop' into feat/#45

jsoonworld committed:
fix(UserController): POST와 GET 메서드 중복 수정 및 코드 리팩토링  - `@PostMapping`과 `@GetMapping`이 중복 사용되었던 부분 수정 - `@RequestBody`와 `@PathVariable`의 변수명 중복 해결 - `FcmTokenReissueRequiredRequest` 객체의 생성 로직 개선

- `@PostMapping`과 `@GetMapping`이 중복 사용되었던 부분 수정
- `@RequestBody`와 `@PathVariable`의 변수명 중복 해결
- `FcmTokenReissueRequiredRequest` 객체의 생성 로직 개선

jsoonworld committed:
feat(UserController) : FCM 토큰 업데이트 API 추가

사용자가 새로 발급받은 FCM 토큰을 서버에 전달하여 업데이트할 수 있는 POST /api/v1/users/{userId}/fcm-tokens 엔드포인트를 추가하였습니다.
- 요청 시 새 토큰을 UserService로 전달해 갱신 처리
- 성공 시 HTTP 201 상태와 FCM_TOKEN_UPDATED 성공 코드를 반환

jsoonworld committed:
feat(FcmTokenUpdateService) : FCM 토큰 업데이트 서비스 인터페이스 정의

사용자의 FCM 토큰을 갱신하기 위한 FcmTokenUpdateService 인터페이스를 추가하였습니다.
- updateFcmToken(Long userId, String newToken) 메서드로 토큰 갱신 책임 부여

jsoonworld committed:
feat(FcmTokenUpdateServiceImpl) : FCM 토큰 업데이트 서비스 구현

UserRepository를 통해 사용자를 조회한 후, FcmTokenUpdateService 인터페이스를 통해 전달받은 새 토큰을 업데이트하는 로직을 구현하였습니다.
- updateFcmToken 메서드에서 존재하지 않는 사용자 ID일 경우 예외 처리
- 트랜잭션 범위 내에서 User 엔티티의 FCM 토큰을 갱신

jsoonworld committed:
feat(UserService) : FCM 토큰 업데이트 로직 추가

새로운 FCM 토큰을 전달받아 유저 객체에 반영할 수 있는 updateFcmToken 메서드를 추가했습니다.
- FcmTokenUpdateService를 활용해 사용자 FCM 토큰 갱신
- @Transactional을 적용하여 DB 변경 사항 반영

jsoonworld committed:
feat(UserSuccessCode) : FCM 토큰 업데이트 성공 코드 추가

FCM 토큰을 새로 갱신했을 때 반환할 FCM_TOKEN_UPDATED 상태 코드를 추가했습니다.
- 201 Created 상태코드와 "FCM 토큰이 성공적으로 업데이트되었습니다." 메시지로 구성

jsoonworld committed:
refactor(User) : FCM 토큰 만료 여부 확인 로직 중복 제거

isFcmTokenExpired 메서드 내 불필요한 두 번의 return 문을 정리하고, currentToken 변수를 사용하도록 수정했습니다.
- 중복된 return 제거
- 코드 가독성과 유지보수성 향상

jsoonworld committed:
feat(FcmToken) : FCM 토큰 값 업데이트 메서드 추가

기존 토큰 객체가 새 값으로 교체될 수 있도록 updateValue 메서드를 추가하였습니다.
- 새로운 FcmToken 인스턴스를 생성해 상태 변경 로직을 캡슐화
- 객체지향적 책임 분리를 강화

jsoonworld committed:
feat(UpdateFcmTokenRequest) : FCM 토큰 업데이트 요청 DTO 추가

FCM 토큰을 새로 발급받아 서버에 전달하기 위한 요청 DTO를 추가했습니다.
- newToken 필드 및 정적 팩토리 메서드 of 정의

jsoonworld committed:
test(FcmTokenUpdateServiceImplTest) : FCM 토큰 업데이트 서비스 단위 테스트 추가

FcmTokenUpdateServiceImpl에 대한 단위 테스트를 작성하였습니다.
- 유효한 사용자 ID와 새로운 토큰으로 요청할 때 정상적으로 업데이트되는지 확인
- 존재하지 않는 사용자 ID에 대해 USER_NOT_FOUND 예외 발생 검증
- null 토큰에 대한 FCM_TOKEN_NOT_NULL 예외 발생 검증

jsoonworld committed:
refactor(FcmConfig) : Firebase 초기화 메시지 및 인코딩 상수 정리

- 초기화 성공/실패 로그 메시지를 FcmLogMessages 상수로 분리하여 재사용성과 일관성 향상
- 서비스 키 문자열 인코딩을 StandardCharsets.UTF_8에서 Encoding.UTF_8로 통일하여 코드 일관성 유지

jsoonworld committed:
refactor(FcmConfig) : FCM 설정 클래스를 공통 config 패키지로 이동

- FcmConfig, FcmProperties 클래스를 기존 user.config 패키지에서 fcm.config 패키지로 이동하여 관심사 분리
- @EnableConfigurationProperties(FcmProperties.class) 위치를 TerningApplication에서 제거하고 FcmConfig로 이동하여 테스트 환경에서의 로딩 오류 방지
- 애플리케이션 초기화 시 Firebase 설정이 공통 모듈로 관리되도록 구조 정리

jsoonworld committed:
merge(messageTemplate): 알림 메시지 템플릿 타입 구현  

[✨ feat] 알림 메시지 템플릿 타입 구현  📷

jsoonworld committed:
merge(message) : 메세지 엔티티 구현

[✨ feat] 메세지 엔티티 구현

jsoonworld committed:
merge(FcmTokenValidate) : FCM 토큰 유효성 검증 API 구현 

[✨ feat] FCM 토큰 유효성 검증 API 구현

jsoonworld committed:
merge(reissueFcmToken) : FCM 토큰 재발급 API 구현 

[✨ feat] FCM 토큰 재발급 API 구현

jsoonworld committed:
refactor(Message) : 메시지 생성 시 포맷 로직을 내부로 이동하여 책임 명확화

- Message.of(template, params) 형태로 개선하여 외부에서 포맷된 메시지를 넘기던 방식 제거
- 메인/서브 메시지의 포맷 책임을 Message 내부로 위임하여 응집도 향상
- 메인 메시지도 확장성 고려해 파라미터 기반 포맷 구조로 통일

jsoonworld committed:
test(Message) : 메시지 생성 방식 변경에 따른 테스트 리팩토링

- Message.of(template, params) 사용 방식에 맞게 테스트 코드 전반 수정
- 메인 메시지도 파라미터 기반 포맷 구조를 사용하므로 null 대신 VALID_PARAMS 전달
- 실패 케이스에서도 Message 내부 포맷 로직을 테스트하도록 변경

junggyo1020 committed:
[✅ chore] 유저 회원가입 성공 시 공통 응답값 처리

junggyo1020 committed:
[✨ feat] 운영서버로부터 회원가입시 유저 정보 받아오는 DTO 구현

junggyo1020 committed:
[✅ chore] 운영서버의 유저ID 저장

junggyo1020 committed:
[✨ feat] 운영서버로부터 회원가입시 통신하는 알림서버 API 구현

junggyo1020 committed:
[✨ feat] 운영서버로부터 회원가입시 통신하는 알림서버 API 구현

jsoonworld committed:
chore(gitignore) : application 설정 파일 및 Q 클래스 디렉토리 추가

- `application.yml`, `application-dev.yml` 파일을 `.gitignore`에 추가하여 민감한 설정 정보가 Git에 포함되지 않도록 처리하였습니다.
- QueryDSL에서 생성되는 Q 클래스 경로(`src/main/generated`)도 무시 대상에 포함시켜 빌드 결과물이 추적되지 않도록 설정했습니다.

jsoonworld committed:
chore(gitignore) : application 설정 파일 및 Q 클래스 디렉토리 추가

- `application.yml`, `application-dev.yml` 파일을 `.gitignore`에 추가하여 민감한 설정 정보가 Git에 포함되지 않도록 처리하였습니다.
- QueryDSL에서 생성되는 Q 클래스 경로(`src/main/generated`)도 무시 대상에 포함시켜 빌드 결과물이 추적되지 않도록 설정했습니다.

jsoonworld committed:
chore(build) : QueryDSL 설정 및 WebFlux, Batch 의존성 추가

- QueryDSL 설정을 위한 의존성 및 관련 디렉토리(`src/main/generated/querydsl/`) 설정을 추가하였습니다.
- `compileOnly` 설정 중복을 제거하고 `querydsl` 설정을 `compileClasspath`에 연결하였습니다.
- WebFlux, Spring Batch 의존성을 프로젝트에 추가하여 관련 기능 개발 준비를 완료했습니다.
- PostgreSQL 의존성 선언 중 중복된 구문을 정리하였습니다.

jsoonworld committed:
feat(QuerydslConfig) : JPAQueryFactory 빈 설정 클래스 추가

- QueryDSL 사용을 위해 `JPAQueryFactory`를 빈으로 등록하는 `QuerydslConfig` 클래스를 추가하였습니다.
- `EntityManager`를 주입받아 `JPAQueryFactory`를 생성하며, 의존성 주입을 통해 QueryDSL 쿼리 객체를 사용할 수 있습니다.
- 해당 설정을 통해 서비스 및 리포지토리 단에서 QueryDSL을 편리하게 활용할 수 있습니다.

jsoonworld committed:
feat(WebClientConfig) : WebClient 설정 클래스 추가

- 외부 API 호출을 위한 `WebClient` 설정 클래스를 추가하였습니다.
- `WebClient.Builder`를 빈으로 주입받아 공용 `WebClient` 인스턴스를 생성합니다.
- 향후 비동기 HTTP 요청 처리 시 재사용 가능한 클라이언트로 활용할 수 있습니다.

jsoonworld committed:
feat(OpsScrapUserResponse) : 운영 서버 스크랩 유저 응답 DTO 추가

- 운영 서버에서 전달받은 스크랩 유저 정보를 표현하기 위한 `OpsScrapUserResponse` record 클래스를 추가하였습니다.
- 유저 ID를 단일 값으로 가지며, 정적 팩토리 메서드 `from`을 통해 명시적인 생성 방식을 제공합니다.
- 향후 알림 서버에서 스크랩 동기화 요청 처리 시 사용됩니다.

jsoonworld committed:
feat(OpsScrapUsersResponse) : 운영 서버 스크랩 유저 목록 응답 DTO 추가

- 운영 서버로부터 수신한 다수의 스크랩 유저 ID를 응답 형태로 감싸는 `OpsScrapUsersResponse` record 클래스를 추가하였습니다.
- 유저 ID 리스트를 `OpsScrapUserResponse`로 변환하는 정적 팩토리 메서드 `of`를 제공합니다.
- 응답 구조를 명확히 분리하여 외부 API 통신 시 일관된 포맷을 유지할 수 있도록 하였습니다.

jsoonworld committed:
feat(OpsScrapUserClient) : 운영 서버로부터 미동기화 유저 목록 조회 기능 구현

- 운영 서버의 `/api/v1/external/scraps/unsynced` API를 호출하여 미동기화된 스크랩 유저 목록을 가져오는 클라이언트를 구현하였습니다.
- WebClient를 사용하여 비동기 방식으로 호출하고, 응답은 `OpsScrapUsersResponse`로 파싱됩니다.
- API URL은 `application.yml`에 설정된 `internal.api.scrap.unsynced` 값을 통해 주입받습니다.
- 향후 배치 작업에서 운영 서버와 동기화를 위한 핵심 의존 컴포넌트로 사용됩니다.

jsoonworld committed:
feat(ScrapController) : 운영 서버로부터 스크랩 유저 동기화 API 구현

- `/api/v1/scraps/sync` POST API를 통해 운영 서버에서 미동기화된 스크랩 유저 정보를 조회하고 저장하는 기능을 구현하였습니다.
- 내부 로직은 `ScrapService`의 `fetchAndSaveScrapUsersFromOps()` 메서드를 통해 처리됩니다.
- 외부 시스템과의 스크랩 동기화를 위한 진입점 역할을 합니다.

jsoonworld committed:
feat(ScrapService) : 운영 서버 스크랩 유저 동기화 서비스 계층 구현

- 운영 서버에서 전달받은 미동기화 스크랩 유저 정보를 처리하기 위한 `ScrapService` 클래스를 추가하였습니다.
- 내부에서는 `ScrapStatusSyncService`를 통해 실제 동기화 로직을 위임 처리하며, 트랜잭션을 적용하여 일관성을 보장합니다.
- 컨트롤러와 도메인 서비스 사이의 연결 역할을 수행하는 계층입니다.

jsoonworld committed:
feat(ScrapStatusSyncService) : 스크랩 유저 동기화 도메인 서비스 인터페이스 정의

- 운영 서버로부터 전달받은 스크랩 유저 정보를 기반으로 동기화 작업을 수행하기 위한 도메인 서비스 인터페이스를 정의하였습니다.
- 향후 구현체에서 동기화 비즈니스 로직을 담당하게 되며, 서비스 계층(`ScrapService`)에서 해당 인터페이스를 통해 로직을 위임합니다.
- 유연한 구현체 교체 및 테스트 용이성을 고려한 구조입니다.

jsoonworld committed:
feat(ScrapStatusSyncServiceImpl) : 운영 서버 스크랩 유저 동기화 로직 구현

- `ScrapStatusSyncService` 인터페이스의 구현체로 `ScrapStatusSyncServiceImpl`을 추가하였습니다.
- `OpsScrapUserClient`를 통해 운영 서버에서 미동기화된 유저 목록을 조회하고, 유저 ID 리스트를 추출합니다.
- 유저 ID 목록이 비어있지 않은 경우 `ScrapSyncManager`를 통해 동기화 처리를 위임합니다.
- 외부 API 연동부터 동기화 실행까지의 전체 프로세스를 담당하는 핵심 로직입니다.

jsoonworld committed:
feat(ScrapSyncManager) : 운영 서버 기반 스크랩 유저 동기화 기능 구현

- 운영 서버에서 전달받은 유저 ID 목록을 기반으로 스크랩 데이터를 생성하거나 상태를 갱신하는 `ScrapSyncManager`를 구현하였습니다.
- 유저 정보와 기존 스크랩 데이터를 Map으로 구성하여 효율적인 조회 및 비교를 수행합니다.
- `createOrUpdateScrap` 메서드로 유저별 처리 로직을 분리하여 가독성과 책임 분리를 강화하였습니다.
- 신규 유저에 대해서는 스크랩을 생성하고, 기존 유저 중 미스크랩 상태인 경우에만 상태를 갱신하도록 처리합니다.

jsoonworld committed:
feat(ScrapUserItemProcessor) : 스크랩 유저 배치 처리용 ItemProcessor 구현

- Spring Batch의 `ItemProcessor`를 구현한 `ScrapUserItemProcessor` 클래스를 추가하였습니다.
- 운영 서버로부터 전달된 스크랩 유저 데이터를 로그로 확인하며, 가공 없이 그대로 다음 단계로 전달합니다.
- 유저 ID 로그를 통해 처리 흐름 추적이 가능하도록 디버깅 메시지를 추가하였습니다.
- 향후 데이터 가공이 필요할 경우 확장 가능성을 염두에 둔 구조입니다.

jsoonworld committed:
feat(ScrapUserItemReader) : 운영 서버에서 미동기화 유저 목록을 읽어오는 배치 리더 구현

- Spring Batch의 `ItemReader`를 구현한 `ScrapUserItemReader` 클래스를 추가하였습니다.
- `OpsScrapUserClient`를 통해 운영 서버에서 미동기화된 스크랩 유저 목록을 조회합니다.
- 최초 read 시 리스트를 조회하고, 이후에는 반복자를 통해 하나씩 처리 대상 유저를 반환합니다.
- 로그를 통해 유저 조회 시작 시점을 명확히 기록하며 배치 흐름을 추적할 수 있도록 하였습니다.

jsoonworld committed:
feat(ScrapUserSyncJobScheduler) : 스크랩 유저 동기화 배치 스케줄러 구현

- 매일 20시 30분에 스크랩 유저 동기화 배치를 실행하는 `ScrapUserSyncJobScheduler`를 구현하였습니다.
- `JobLauncher`를 통해 `scrapSyncJob`을 실행하며, 매 실행마다 고유한 `run.id` 파라미터를 부여합니다.
- 성공/실패 여부에 따라 로그를 출력하여 모니터링이 가능하도록 구성하였습니다.
- 배치 실행 스케줄은 cron 표현식 `"0 30 20 * * *"`로 설정되어 있으며, 운영 환경 기준에 따라 변경 가능합니다.

jsoonworld committed:
feat(ScrapUserSyncJobConfig) : 스크랩 유저 동기화 배치 Job 및 Step 설정 추가

- Spring Batch 기반의 스크랩 유저 동기화 배치 잡을 구성하였습니다.
- `scrapUserSyncJob`은 단일 Step으로 구성되며, Reader → Processor → Writer 순으로 유저 동기화를 처리합니다.
- chunk 단위는 100으로 설정되어 있으며, 장애 복원력을 위해 예외 발생 시 최대 3회까지 재시도하도록 구성하였습니다.
- Step 이름에 오타(`"s crapUserSyncStep"`)가 있어 `"scrapUserSyncStep"`으로의 수정이 필요합니다.

jsoonworld committed:
feat(ScrapSyncWriter) : 스크랩 유저 배치 동기화 ItemWriter 구현

- Spring Batch의 `ItemWriter`를 구현한 `ScrapSyncWriter` 클래스를 추가하였습니다.
- 배치 처리 중 전달받은 유저 ID를 기반으로 `ScrapSyncManager`를 통해 스크랩 동기화를 수행합니다.
- 로그를 통해 chunk 내 처리 대상 유저 수를 기록하여 배치 모니터링이 가능하도록 구성하였습니다.
- 중복 유저 ID는 `distinct()` 처리하여 동기화 요청을 최소화합니다.

jsoonworld committed:
feat(ScrapRepository) : 스크랩 도메인 JPA 리포지토리 인터페이스 추가

- `Scrap` 엔티티에 대한 기본 CRUD 및 커스텀 쿼리 확장을 위한 `ScrapRepository` 인터페이스를 정의하였습니다.
- Spring Data JPA의 `JpaRepository`와 사용자 정의 인터페이스 `ScrapRepositoryCustom`을 함께 상속받아 구현 유연성을 확보하였습니다.

jsoonworld committed:
feat(ScrapRepositoryCustom) : 유저 ID 목록 기반 스크랩 조회 메서드 정의

- 스크랩 도메인에서 유저 ID 목록을 기준으로 해당 유저들의 스크랩 정보를 조회하기 위한 커스텀 리포지토리 인터페이스를 정의하였습니다.
- `findScrapsByUserIds` 메서드를 통해 배치 동기화 작업에서 활용 가능한 쿼리를 구현할 수 있도록 구조를 분리하였습니다.
- QueryDSL 기반 구현체(`ScrapRepositoryImpl`)에서 해당 메서드의 실제 로직이 작성될 예정입니다.

jsoonworld committed:
feat(ScrapRepositoryImpl) : 유저 ID 목록으로 스크랩 조회하는 커스텀 쿼리 구현

- `ScrapRepositoryCustom`에 정의된 `findScrapsByUserIds` 메서드의 구현체를 추가하였습니다.
- QueryDSL의 `JPAQueryFactory`를 사용하여 유저 ID 목록을 기반으로 해당 유저들의 스크랩 정보를 조회합니다.
- 배치 처리나 동기화 작업 등에서 유저별 스크랩 상태를 빠르게 조회할 수 있도록 효율적인 쿼리를 구성하였습니다.

jsoonworld committed:
feat(TerningApplication) : 배치 처리 활성화를 위한 @EnableBatchProcessing 추가

- Spring Batch 기능을 사용하기 위해 `@EnableBatchProcessing` 애노테이션을 `TerningApplication` 클래스에 추가하였습니다.
- 해당 설정으로 배치 Job, Step 구성 클래스에서 필요한 Batch 인프라 빈들이 자동 등록됩니다.
- 기존의 JPA Auditing 기능(`@EnableJpaAuditing`)과 함께 동작하도록 구성하였습니다.

jsoonworld committed:
feat(UserRepository) : 커스텀 리포지토리 연동을 위한 인터페이스 확장

- 유저 도메인에서 커스텀 쿼리 작성을 위해 `UserRepositoryCustom` 인터페이스를 `UserRepository`에 추가하였습니다.
- Spring Data JPA의 기본 CRUD 기능과 함께 커스텀 조회 로직 구현이 가능하도록 구조를 확장하였습니다.
- 동기화 대상 유저 조회 등 복잡한 쿼리를 커스텀 리포지토리에서 처리할 수 있도록 대비한 작업입니다.

jsoonworld committed:
feat(UserRepositoryCustom) : 유저 ID 목록 기반 유저 조회 메서드 정의

- 유저 ID 리스트를 기반으로 유저 정보를 조회하고, 결과를 ID-Keyed Map으로 반환하는 커스텀 메서드를 정의하였습니다.
- 대량의 유저 데이터를 효율적으로 조회하고, 후속 로직에서 빠르게 접근할 수 있도록 Map 구조를 사용합니다.
- 배치 동기화나 대용량 처리 상황에서 반복 조회를 줄이고 성능을 개선하기 위한 기반 메서드입니다.

jsoonworld committed:
feat(UserRepositoryImpl) : 유저 ID 목록 기반 Map 조회 커스텀 쿼리 구현

- `UserRepositoryCustom`에 정의된 `findUsersByIds` 메서드의 구현체를 추가하였습니다.
- QueryDSL을 사용하여 유저 ID 리스트로 유저 목록을 조회하고, 결과를 ID-Keyed Map으로 변환합니다.
- 후속 로직에서 유저 정보를 빠르게 조회할 수 있도록 성능을 고려한 구조입니다.
- 대량 처리나 배치 동기화에서 효율적인 데이터 접근이 가능해집니다.

jsoonworld committed:
test(ScrapStatusSyncServiceImpl) : 스크랩 상태 동기화 서비스 단위 테스트 작성

- `ScrapStatusSyncServiceImpl`에 대한 단위 테스트 클래스를 작성하였습니다.
- 가짜 클라이언트(`FakeOpsScrapUserClient`)와 인메모리 테스트 리포지토리를 사용하여 실제 의존성을 제거한 상태에서 테스트합니다.
- 스크랩이 없는 유저를 새로 동기화하는 경우, 이미 스크랩된 유저는 건너뛰는 경우 등 정상 동작을 검증합니다.
- 운영 서버에서 유저가 없거나, 전달된 유저가 존재하지 않는 경우 예외 없이 무시되는 실패 케이스도 포함되어 있습니다.

jsoonworld committed:
test(ScrapSyncManager) : 스크랩 동기화 매니저 단위 테스트 작성

- `ScrapSyncManager`의 핵심 동기화 로직에 대한 단위 테스트를 작성하였습니다.
- 스크랩이 존재하지 않는 유저에 대한 신규 저장, UNSCRAPPED 상태의 SCRAPPED 전환, 이미 SCRAPPED 상태일 경우 무시 등의 정상 동작을 검증합니다.
- 유저가 존재하지 않는 경우, 중복된 유저 ID가 들어온 경우 등 예외 케이스에 대한 방어 로직도 테스트에 포함되었습니다.
- 인메모리 리포지토리를 활용해 외부 의존성 없이 순수하게 도메인 로직 검증이 가능하도록 구성하였습니다.

jsoonworld committed:
test(ScrapSyncWriter) : 스크랩 유저 배치 Writer 단위 테스트 작성

- 배치에서 유저 ID chunk를 처리하는 `ScrapSyncWriter`에 대한 단위 테스트를 작성하였습니다.
- SCRAPPED 상태가 아닌 유저에 대해 새로 스크랩을 저장하거나, 상태를 갱신하는지 검증합니다.
- 이미 SCRAPPED된 유저는 건너뛰는 로직, 존재하지 않는 유저 ID는 무시하는 로직도 포함하여 방어적 처리를 확인합니다.
- 중복된 유저 ID 입력 시 한 번만 처리되도록 검증하여 데이터 무결성을 확인합니다.
- `Chunk` 객체를 직접 생성하여 Spring Batch 없이도 배치 Writer 로직을 테스트 가능하도록 구성하였습니다.

jsoonworld committed:
test(ScrapRepositoryTest) : 인메모리 스크랩 리포지토리 테스트 구현체 구현

- 테스트 환경에서 사용하기 위한 `ScrapRepository` 및 `ScrapRepositoryCustom`의 인메모리 구현체를 작성하였습니다.
- 실제 DB 없이도 도메인 로직 및 배치 테스트가 가능하도록 `List` 기반 저장소와 ID 자동 증가 기능(`AtomicLong`)을 구현했습니다.
- `save`, `saveAll`, `findScrapsByUserIds`, `deleteById`, `findAll` 등 테스트에 필요한 핵심 메서드만 구현하였습니다.
- 나머지 `JpaRepository` 기본 메서드는 실제 테스트 목적에 따라 점진적으로 구현 가능합니다.

jsoonworld committed:
test(UserRepositoryTest) : 커스텀 메서드 포함한 인메모리 유저 리포지토리 구현

- `UserRepositoryCustom`을 구현하도록 `UserRepositoryTest`를 확장하였습니다.
- `findUsersByIds` 메서드를 구현하여 유저 ID 목록에 대응하는 유저들을 Map 형태로 반환할 수 있도록 구성하였습니다.
- 테스트에서 DB 없이 도메인 및 배치 로직을 검증할 수 있도록 인메모리 기반으로 동작합니다.
- 기존 저장 로직과 함께 다양한 시나리오 테스트를 지원하기 위한 기반을 마련하였습니다.

jsoonworld committed:
feat(MessageTemplateType) : 메시지 전송 일정 관리 기능 구현

SendSchedulePolicy를 MessageTemplateType에 추가하여 각 템플릿의 전송 요일 및 시간을 관리할 수 있도록 기능을 확장했습니다.

- 메시지 템플릿별 전송 여부 확인: isTodayScheduled(now)
- 다음 전송 시점 계산: nextScheduledDate(now)
- 대상 일치 여부 확인 메서드 추가: matchesTarget(target)
- 필드명 messageTargetType으로 명확화

jsoonworld committed:
feat(SendSchedulePolicy) : 템플릿별 전송 요일 및 시간 정책 정의

MessageWeeklyPolicy를 기반으로 각 템플릿의 전송 요일과 시간을 정의하는 SendSchedulePolicy enum을 구현했습니다.

- INTERESTED_ANNOUNCEMENT: 매주 월요일 21시
- INTERNSHIP_RECOMMENDATION: 매주 수요일 20시 30분
- TRENDING_INTERNSHIP: 매주 금요일 18시

isDue(now), nextScheduleAfter(now) 메서드를 통해 현재 시간 기준 전송 여부 및 다음 전송 시점을 판단할 수 있습니다.

jsoonworld committed:
refactor(Message) : 필드 및 생성자 명칭을 의미에 맞게 정리

기존 formattedMainMessage, formattedSubMessage 필드명을 main, sub으로 변경하여 의미를 단순화하고 명확히 표현했습니다.

- 생성자 및 정적 팩토리 메서드 파라미터명 수정
- isSameType 메서드 파라미터명도 의미에 맞게 other로 변경

불필요한 형용사 제거를 통해 도메인 모델의 가독성과 표현력을 개선했습니다.

jsoonworld committed:
feat(MessageWeeklyPolicy) : 메시지 전송 요일 및 시간 정책 VO 구현

요일(DayOfWeek)과 시간(LocalTime)을 조합하여 메시지 전송 일정을 표현하는 Value Object(MessageWeeklyPolicy)를 구현했습니다.

- of() 정적 팩토리 메서드 제공
- isTodaySchedule(now): 현재 시각이 설정된 전송 일정과 일치하는지 확인
- nextScheduleDate(now): 다음 전송 일정을 계산

SendSchedulePolicy의 일정 계산을 책임지는 VO로 설계되었습니다.

jsoonworld committed:
refactor(NotificationController) : 알림 생성 요청 DTO를 NotificationCreateRequest로 변경

알림 생성 API의 요청 본문을 NotificationCreateRequest로 수정하여 의미를 명확하게 표현했습니다.

- 기존 GenerateNowRequest → NotificationCreateRequest로 교체
- createNotification 메서드 호출 시 새로운 DTO 사용

DTO 명명 정비를 통해 역할과 목적이 명확하게 드러나도록 개선했습니다.

jsoonworld committed:
feat(NotificationCommandService) : 알림 생성 도메인 명령 서비스 인터페이스 정의

알림 생성을 위한 도메인 계층 명령 서비스 인터페이스 NotificationCommandService를 정의했습니다.

- create(request, scheduledTime): 알림 생성 책임을 명확히 분리
- NotificationService의 역할을 세분화하여 도메인 주도 설계 방식에 부합

도메인 내부 로직을 명확히 분리하고 책임을 위임하기 위한 구조 개선입니다.

jsoonworld committed:
feat(NotificationCommandServiceImpl) : 알림 생성 명령 서비스 구현체 작성

NotificationCommandService의 구현체로 NotificationCommandServiceImpl을 추가했습니다.

- 요청에서 템플릿 타입을 추출하여 MessageTemplateType으로 변환
- NotificationWriter를 통해 알림 생성 책임을 위임
- ScheduledTime을 기반으로 전송 일정 지정

알림 생성을 도메인 명령 흐름에 맞춰 처리하도록 구조화했습니다.

jsoonworld committed:
feat(NotificationCreationExecutor) : 알림 생성 실행 책임 인터페이스 정의

알림 생성을 실제로 실행하는 책임을 가지는 NotificationCreationExecutor 인터페이스를 정의했습니다.

- execute(template, scheduledTime): 템플릿과 스케줄 정보를 받아 알림 생성 실행
- 실행 책임을 분리함으로써 전략 및 실행 주체를 유연하게 확장 가능

알림 생성 흐름의 명확한 역할 분리를 위한 설계입니다.

jsoonworld committed:
feat(NotificationCreationExecutorImpl) : 알림 생성 실행 로직 구현

NotificationWriter를 활용하여 템플릿과 스케줄 정보를 기반으로 알림을 생성하는 로직을 구현했습니다.

jsoonworld committed:
feat(NotificationService) : 알림 생성 및 실행 서비스 구현

NotificationCommandService와 NotificationCreationExecutor를 통해 알림 생성 요청 처리 및 실행 기능을 제공하는 NotificationService를 구현했습니다.

jsoonworld committed:
feat(NotificationWriter) : 알림 저장 기능 인터페이스 정의

템플릿과 전송 시간을 기반으로 알림을 저장하는 기능을 추상화한 NotificationWriter 인터페이스를 정의했습니다.

jsoonworld committed:
feat(NotificationWriterImpl) : 알림 저장 로직 구현체 작성

NotificationWriter 인터페이스의 구현체로 NotificationWriterImpl을 작성했습니다.

- 대상 타입에 따라 유저 조회 로직 분기
- 사용자별 메시지 생성 및 알림 객체 구성
- 일괄 저장 처리로 알림 등록

상수 USERNAME을 통해 메시지 포맷팅 키를 명확하게 관리합니다.

jsoonworld committed:
feat(NotificationCreationTasklet) : 배치용 알림 생성 태스크릿 구현

MessageTemplateType의 모든 템플릿을 순회하며 알림을 생성하는 NotificationCreationTasklet을 구현했습니다.

- 현재 시각 기준 ScheduledTime 생성
- NotificationService를 통해 각 템플릿에 대한 알림 실행
- 배치 Step에서 사용 가능한 Tasklet 형태로 구성

jsoonworld committed:
feat(NotificationScheduler) : 알림 생성 배치 스케줄러 구현

매일 새벽 3시에 알림 생성 배치를 실행하는 NotificationScheduler를 구현했습니다.

- cron 표현식으로 스케줄 설정: 0 0 3 * * *
- timestamp 기반 JobParameters 생성
- JobLauncher를 통해 notificationCreationJob 실행

jsoonworld committed:
feat(NotificationJobConfig) : 알림 생성 배치 잡 및 스텝 설정

알림 생성을 위한 배치 잡(notificationCreationJob)과 태스크릿 기반 스텝(notificationCreationStep)을 설정했습니다.

- NotificationCreationTasklet을 스텝에 등록
- JobRepository 및 트랜잭션 매니저를 활용한 설정 구성

jsoonworld committed:
feat(NotificationErrorCode) : 알림 도메인 에러 코드 정의

알림 도메인에서 발생할 수 있는 예외 상황에 대응하기 위한 NotificationErrorCode enum을 정의했습니다.

- INVALID_TARGET_TYPE
- INVALID_TEMPLATE_TYPE

공통 ErrorCode 인터페이스를 구현하고, 메시지에 [NOTIFICATION ERROR] 접두어를 추가했습니다.

jsoonworld committed:
feat(NotificationException) : 알림 도메인 전용 예외 클래스 구현

NotificationErrorCode를 기반으로 하는 알림 도메인 전용 예외 클래스 NotificationException을 구현했습니다.

- BaseException 상속
- 에러 코드 기반 예외 처리 일관성 확보

jsoonworld committed:
feat(NotificationsRepository) : 알림 저장소 인터페이스 생성

Notification 엔티티를 관리하기 위한 JPA 저장소 인터페이스 NotificationsRepository를 생성했습니다.

jsoonworld committed:
feat(ScheduledTime) : 알림 전송 시각 VO 구현

알림 전송 시각을 표현하는 값 객체 ScheduledTime을 구현했습니다.

- of(LocalDateTime) 정적 팩토리 메서드 제공
- isDue(now): 지정된 시각이 현재 시각 이전인지 판단
- JPA @Embeddable 및 불변 객체 설계 적용

jsoonworld committed:
feat(SentStatus) : 알림 발송 여부 VO 구현

알림의 발송 상태를 나타내는 값 객체 SentStatus를 구현했습니다.

- markNotSent(): 초기 상태
- markSent(): 발송 완료 상태로 전환
- isSent(), value(): 발송 여부 확인

JPA @Embeddable 적용 및 불변 객체 패턴을 기반으로 설계했습니다.

jsoonworld committed:
feat(NotificationCreateRequest) : 알림 생성 요청 DTO 정의

알림 생성을 위한 요청 데이터를 담는 DTO인 NotificationCreateRequest를 정의했습니다.

- 템플릿 타입 전달을 위한 template 필드 포함
- 정적 팩토리 메서드 of(String) 제공

jsoonworld committed:
refactor(ScrapUserSyncJobScheduler) : 상수 분리 및 예외 처리 개선

- CRON 스케줄, run.id 키 값을 상수로 추출하여 재사용성 향상
- 불필요한 로그 상수 제거 및 log 메시지 통일
- 예외 발생 시 ScrapException을 던지도록 처리 방식 개선

jsoonworld committed:
feat(ScrapErrorCode) : 스크랩 동기화 배치 실패 에러 코드 추가

스크랩 동기화 배치 실행 중 예외 상황을 처리하기 위해 SYNC_JOB_FAILED 에러 코드를 추가했습니다.

jsoonworld committed:
feat(ScrapRepositoryCustom) : 스크랩 유저 조회 메서드 정의

스크랩된 유저 조회 기능 강화를 위해 ScrapRepositoryCustom에 사용자 기반 조회 메서드를 정의했습니다.

- findScrapsByUserIds(List<Long>)
- findDistinctScrappedUsers()

jsoonworld committed:
feat(ScrapRepositoryImpl) : 스크랩된 유저 조회 쿼리 추가

스크랩 상태가 SCRAPPED인 유저만 조회하는 findDistinctScrappedUsers 메서드를 ScrapRepositoryImpl에 구현했습니다.

- distinct + fetchJoin으로 중복 제거 및 성능 최적화

jsoonworld committed:
feat(Notifications) : 알림 관리 일급 컬렉션 구현

Notifications 클래스에 불변성을 보장하는 일급 컬렉션을 구현했습니다.

- `add()`, `remove()` 메서드로 새로운 알림 추가 및 제거 시 불변 컬렉션 반환
- `values()` 메서드로 알림 리스트를 불변 리스트로 반환
- `empty()` 메서드를 통해 빈 알림 목록을 생성할 수 있도록 함

jsoonworld committed:
feat(Scraps) : 스크랩 관리 일급 컬렉션 구현

Scraps 클래스에 불변성을 보장하는 일급 컬렉션을 구현했습니다.

- `add()`, `remove()` 메서드로 새로운 스크랩 추가 및 제거 시 불변 컬렉션 반환
- `values()` 메서드로 스크랩 리스트를 불변 리스트로 반환
- `empty()` 메서드를 통해 빈 스크랩 목록을 생성할 수 있도록 함

jsoonworld committed:
fix(User) : 중복된 notifications와 scraps 필드 제거

User 엔티티에서 중복된 notifications와 scraps 필드를 제거하고, 일급 컬렉션을 사용하도록 수정했습니다.

- notifications 필드: List<Notifications> → List<Notification>으로 변경
- scraps 필드: List<Scrap> → 일급 컬렉션인 Scraps로 변경
- 불필요한 필드 중복을 해결하여 가독성과 코드 안정성 향상

jsoonworld committed:
test(MessageTest) : 메시지 포맷 검증 테스트 추가

- 메시지 포맷에서 {username}이 포함되지 않도록 검증하는 테스트 추가

jsoonworld committed:
test(NotificationWriterImplTest) : 알림 작성 서비스 테스트 추가

- 모든 유저에게 '최근 인턴십 추천' 알림 생성 테스트 구현
- 스크랩한 유저에게 '관심 공고 알림' 생성 테스트 구현
- 스크랩하지 않은 유저에게 알림이 생성되지 않도록 검증하는 테스트 구현
- 유저가 없는 경우 알림이 생성되지 않음을 검증하는 테스트 구현

jsoonworld committed:
feat(NotificationsRepositoryTest) : NotificationsRepository 테스트 구현

- Notification 엔티티의 저장, 조회, 삭제 기능을 모킹하여 테스트하는 Repository 구현
- JPA의 기본 메서드들에 대한 구현 추가 (save, findById, delete 등)
- 테스트 환경에서 사용할 수 있는 메모리 기반 Repository 구현

jsoonworld committed:
feat(ScrapRepositoryTest) : ScrapRepository 테스트 구현

- Scrap 엔티티의 저장, 조회, 삭제 기능을 모킹하여 테스트하는 Repository 구현
- 스크랩 상태가 'SCRAPPED'인 유저를 구별하는 findDistinctScrappedUsers 메서드 구현
- JPA의 기본 메서드들을 모킹하여 테스트 환경에서 사용할 수 있도록 Repository 구현

jsoonworld committed:
refactor(NotificationCreationTasklet) : ScheduledTime 호출 개선

junggyo1020 committed:
[✅ chore] 시큐리티 의존성 제거

jsoonworld committed:
merge(message) : 메시지 포맷 로직을 내부로 이동하여 생성 책임 명확화 및 응집도 개선 

[♻️ refactor] 메시지 포맷 로직을 내부로 이동하여 생성 책임 명확화 및 응집도 개선

jsoonworld committed:
merge(scrap) :  운영서버에서 스크랩 유저 조회 후 저장하는 동기화 로직 구현

[✨ feat] 운영서버에서 스크랩 유저 조회 후 저장하는 동기화 로직 구현

jsoonworld committed:
merge(notificationMessage) : 메시지-유저 조합을 통해 알림 생성 및 저장 배치 구현 

[✨ feat] 메시지-유저 조합을 통해 알림 생성 및 저장 배치 구현

jsoonworld committed:
Merge branch 'feat/#52' into feat/#47-after

jsoonworld committed:
Merge branch 'develop' into feat/#47-after

jsoonworld committed:
merge(fix) : [🔀 merge] 브렌치들간 충돌 해결

[🔀 merge] 브렌치들간 충돌 해결

junggyo1020 committed:
Merge branch 'feat/#49-before' into develop

# Conflicts:
#	src/main/java/org/terning/fcm/config/FcmConfig.java
#	src/main/java/org/terning/user/api/UserController.java
#	src/main/java/org/terning/user/application/UserService.java
#	src/main/java/org/terning/user/common/success/UserSuccessCode.java

jsoonworld committed:
feat(ScrapController) : 스크랩 동기화 결과 처리 메서드 구현

- `handleScrapSyncResult` 메서드를 구현하여 외부에서 전송된 스크랩 동기화 결과를 처리
- `ScrapUserIdsRequest`를 요청 본문으로 받아 처리 후, 정상 응답을 반환하는 기능 추가

jsoonworld committed:
refactor(OpsScrapUserResponse, OpsScrapUsersResponse) : 패키지 이동

jsoonworld committed:
refactor(OpsScrapUserClient) : Ops 스크랩 유저 클라이언트 패키지 이동 및 API 경로 수정

- `OpsScrapUserClient` 클래스를 새로운 `scrap.config` 패키지로 이동하여 구성
- `ops.server.url` 값을 `@Value` 어노테이션을 사용해 프로퍼티 파일에서 읽어오도록 변경
- 스크랩 유저 조회 및 동기화 결과 알림 전송을 처리하는 메서드 `fetchUnsyncedUsers` 및 `notifyScrapResult`를 구현하여 OPS 서버와 통신하는 기능 추가

jsoonworld committed:
fix(ScrapStatusSyncServiceImpl) : 패키지 충돌 해결 및 클라이언트 사용 수정

- `OpsScrapUserClient` 클래스의 패키지 충돌을 해결하기 위해 임포트 경로를 수정
- `scrap.config.OpsScrapUserClient`를 올바르게 사용하도록 수정하여 외부 API와 통신하는 기능 구현
- `opsScrapUserClient.notifyScrapResult(userIds)` 메서드 호출을 통해 동기화 결과를 알림

jsoonworld committed:
chore(OpsScrapUserClient,OpsScrapUserResponse) : 임포트 경로 수정

jsoonworld committed:
feat(ScrapUserIdsRequest) : 스크랩 유저 ID 요청 DTO 추가

- `ScrapUserIdsRequest` 레코드를 추가하여 스크랩 유저 ID 목록을 담는 요청 DTO 구현
- `of` 메서드를 사용하여 리스트를 받아 DTO 인스턴스를 생성할 수 있도록 추가
- 클라이언트와 API 통신에서 유저 ID 목록을 전송하는 데 사용

jsoonworld committed:
merge(ScrapController) : 운영서버-알림서버 동기화 응답 반환

[✨ feat] 운영서버-알림서버 동기화 응답 반환

junggyo1020 committed:
Merge remote-tracking branch 'origin/develop' into develop

junggyo1020 committed:
[✅ chore] 브랜치 간 충돌 해결

junggyo1020 committed:
Update Dockerfile

junggyo1020 committed:
Update DOCKER-CD.yml

junggyo1020 committed:
Update DOCKER-CD.yml

junggyo1020 committed:
Update DOCKER-CD.yml

junggyo1020 committed:
Update DOCKER-CD.yml

junggyo1020 committed:
Update DOCKER-CD.yml

junggyo1020 committed:
Update DOCKER-CD.yml

junggyo1020 committed:
Update DOCKER-CD.yml

junggyo1020 committed:
Update DOCKER-CD.yml

junggyo1020 committed:
Update DOCKER-CD.yml

junggyo1020 committed:
[✨ feat] 긴 JSON 환경변수를 JSON 파일로 전달후 mount하는 방식으로 변경

junggyo1020 committed:
Update DOCKER-CD.yml

junggyo1020 committed:
Update DOCKER-CD.yml

junggyo1020 committed:
Update DOCKER-CD.yml

junggyo1020 committed:
Update DOCKER-CD.yml

junggyo1020 committed:
[✨ feat] 긴 JSON 환경변수 인코딩방식 변경

junggyo1020 committed:
[✨ feat] 긴 JSON 환경변수 인코딩방식 변경

junggyo1020 committed:
Update DOCKER-CD.yml

junggyo1020 committed:
Update DOCKER-CD.yml

jsoonworld committed:
Update DOCKER-CD.yml

jsoonworld committed:
Update DOCKER-CD.yml

jsoonworld committed:
Update DOCKER-CD.yml

jsoonworld committed:
Update DOCKER-CD.yml

jsoonworld committed:
Update DOCKER-CD.yml
